composables/api/useGetUserDetail/index.ts

```diff
@@ -10,6 +14,7 @@ export async function useGetUserDetail(
   const { onError, onOkay } = opts
   const responseData = ref<UserViewModel>({
     id: '',
+    role: 'Banned',
```

Comment:
Why is Banned the default?

Reply:
Empty strings aren't allow based on the type of the property, so by default I assumed `Banned` would be best since it would act like a user is not signed in

Reply:
That feels like a hack. Is it possible to just use an empty object for the whole viewmodel instead? Maybe there's some other pattern that would work here.

Reply:
This is a great example of where typescript hurts more than it helps. You should just be able to initialize it with a null value and then assume that it's not null if the fetch operation succeeds.

Reply:
What's the update on this? I believe there was a discussion in chat, right?

Reply:
We could just add an empty string to `UserRole`, tbh. We would know in-code that we're in a logged out state, and design accordingly.